designs/containers/vnf-deployment/vnf-deployment.md

```diff
@@ -0,0 +1,106 @@
+# OSC Container Support - VNF Deployment
+This document describes the design changes needed for OSC to support deployments of container VNFs on a Kubernetes(K8s) cluster.  The deployed VNF instances will be used to protect the network traffic between workload containers present on the cluster.  
+
+## Assignees
+Emanoel Xavier - https://github.com/emanoelxavier
+
+## Background
+OSC currently supports protection of workloads hosted on VM based virtualized environments. This work expands OSC to also control security on container based environments orchestrated by [Kubernetes](#kubernetes-home). The adoption of container technologies is becoming more and more widespread due to its benefits such as easy maintainability, reusability, and minimal overhead with K8s being one of the most popular container orchestrators.  
+For the first release of the OSC containers support the focus will be mostly on the integration points between OSC, the virtualization environment (K8s) and the software defined network(SDN) controller services needed to discovering the workloads to be protected, deploying the security VNFs and performing traffic redirection through a demonstrable E2E flow.  
+
+## Constraints and Assumptions
+
+### OVN with SFC for Kubernetes
+To perform the redirection through a container VNF using OVN the container must be on the same Kubernetes node as the protected pod. For this reason OSC may need to deploy one pod (container) per node in the cluster. There may be a way to configure a Deployment in K8s to indicate every node must have a given container. If not, this might still be possible with a single deployment: OSC can create one deployment with multiple pod specs using node selectors. Each spec would select an specific node. This approach requires further investigation and must be addressed when the integration with OVN is planned.  
+
+### Nuage SDN for Kubernetes
+Nuage does not present a restriction with respect to where the container VNF should be deployed. 
+
+
+## Design Changes
+The implementation of this feature will require changes on the OSC entities and operations involved on the definition and registration of VNF images; deployment specification; and monitoring of container VNFs. 
+
+### Container VNF Images
+In order for a user to deploy a VM VNF using OSC they must register the image file of the VNF along with some additional metadata. For container VNFs this model will be further simplified.  The image for the container is not necessary as it will be retrieved by the virtualization platform from a Docker registry, the user will simply need to provide the metadata needed for OSC to match the VNF with the supported security manager and virtualization platform versions and create the [K8s Deployment object](#kubernetes-deployment).

```

Comment:
I believe the typical deployment model for k8s and docker is to pull the image from a registry. If not public then a private registry. We could add that functionality to OSC but If we do that I think the correct way would be to add the registry functionality to it, in other words k8s/docker would call OSC to pull the image like they do for other docker registries.  Another thing to consider is if customers would have already another private registry where they maintain other docker images, if that is not the case then extending OSC to do this might be a good thing.